src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -626,6 +644,9 @@ internal unsafe struct MethodFixupCell
             public IntPtr MethodName;
             public ModuleFixupCell* Module;
             private int Flags;
+#if TARGET_WINDOWS && TARGET_X86
+            public int SignatureBytes;
```

Comment:
You can use upper 16 bits in Flags to store the argument count. (stdcall is limited to 64k argument - `ret N` instruction cannot handle more.)

Reply:
Makes sense, I was thinking about doing that myself but was not sure about the limits.